extract_indices.ipynb

```diff
@@ -0,0 +1,366 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "eras = [\n",
+    "    \"2011-2040\",\n",
+    "    \"2041-2070\",\n",
+    "    \"2071-2100\",\n",
+    "]\n",
+    "\n",
+    "# list of the index variable names available from config\n",
+    "index_list = [name for index_list in idx_varname_lu.values() for name in index_list]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "428c0934-f1ff-49a3-b585-5056288a50b4",
+   "metadata": {},
+   "source": [
+    "Iterate! Iterate! Loop over all possibilities and populate the excel sheet, inefficently but straightforwardly:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "948682a2-97c2-4082-b1d1-da756dc30a84",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Kaktovik done\n",
+      "Stevens Village done\n",
+      "Igiugik Village done\n",
+      "Levelock done\n",
+      "Eyak done\n",
+      "Ketchikan done\n",
+      "Aleutians done\n"
+     ]
+    }
+   ],
+   "source": [
+    "with pd.ExcelWriter(idx_era_summary_fp, engine=\"openpyxl\") as writer:\n",
+    "    dfs = []\n",
+    "    for location in locations:\n",
+    "        lat, lon = locations[location]\n",
+    "        df_rows = []\n",
+    "        for era in eras:\n",
+    "            start_year, end_year = era.split(\"-\")\n",
+    "            for model in ds.model.values:\n",
+    "                for scenario in scenarios:\n",
+    "                    # we aren't looking at any historical eras here.\n",
+    "                    if scenario == \"hist\":\n",
+    "                        continue\n",
+    "                    for index in index_list:\n",
+    "                        da = ds[index].sel(\n",
+    "                            model=model,\n",
+    "                            scenario=scenario,\n",
+    "                            year=slice(int(start_year), int(end_year))\n",
+    "                        ).sel(\n",
+    "                            lat=lat,\n",
+    "                            lon=lon,\n",
+    "                            method=\"nearest\"\n",
+    "                        )\n",
+    "                        # for aggr_var in aggr_var_lu:\n",
+    "\n",
+    "                        df_rows.append({\n",
+    "                            \"model\": model,\n",
+    "                            \"scenario\": scenario,\n",
+    "                            \"era\": era,\n",
+    "                            \"idx_var\": index,\n",
+    "                            \"min\": np.nanmin(da.values).round(1),\n",
+    "                            \"mean\": np.nanmean(da.values).round(1),\n",
+    "                            \"max\": np.nanmax(da.values).round(1),\n",
+    "                        })\n",
+    "\n",
+    "        # create dataframe write dataframe to a sheet in the excel file   \n",
+    "        df = pd.DataFrame(df_rows)\n",
```

Comment:
It looks like converting `df_rows` to a DataFrame ends up undoing the rounding that was done previously. Looks like this can be fixed by rounding the DataFrame after conversion like this:

```
pd.DataFrame(df_rows).round(1)
```